agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/processors/ExporterWithLogProcessor.java

```diff
@@ -0,0 +1,65 @@
+package com.microsoft.applicationinsights.agent.internal.processors;
+
+import com.microsoft.applicationinsights.agent.internal.processors.AgentProcessor.IncludeExclude;
+import com.microsoft.applicationinsights.agent.internal.wasbootstrap.configuration.Configuration.ProcessorConfig;
+import com.microsoft.applicationinsights.customExceptions.FriendlyException;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+public class ExporterWithLogProcessor implements SpanExporter {
+
+    private final SpanExporter delegate;
+    private final SpanProcessor logProcessor;
+
+    // caller should check config.isValid before creating
+    public ExporterWithLogProcessor(ProcessorConfig config, SpanExporter delegate) throws FriendlyException {
+        config.validate();
+        logProcessor = SpanProcessor.create(config);
+        this.delegate = delegate;
+    }
+
+    @Override
+    public CompletableResultCode export(Collection<SpanData> spans) {
+        // we need to filter attributes before passing on to delegate
+
+            List<SpanData> copy = new ArrayList<>();
```

Comment:
let's not reformat the whole class yet, since we have two branches going on right now (`aad` and `main`) so could cause merge conflicts

i've created a [work item](https://msazure.visualstudio.com/One/_workitems/edit/10033056) to reformat all code using same format used in OTel Java projects